data/mesongenbuildreq.py

```diff
@@ -0,0 +1,16 @@
+import subprocess
+import json
+import sys
+deps_json = json.loads(subprocess.run([sys.argv[1], "introspect", "--dependencies", "meson.build"], capture_output=True).stdout)
+deps = dict(sorted(zip([x['name'] for x in deps_json],[x['version'] for x in deps_json])))
+deps.pop('', None)
+for lib, versions in deps.items() :
+     # Prepare version constraint
+     version_str = ' ' + ' '.join(versions) if versions else ''
+     line = []
+     for prefix in ["cmake", "pkgconfig", "qmake"] :
+         buildreq = (f"{prefix}({lib}){version_str}")
+         if buildreq.split('=')[-1] == '' and '=' in buildreq :
+             buildreq = buildreq.split('=')[0]
+         line.append(buildreq)
+     print(f"BuildRequires: ({' or '.join(line)})")
```

Comment:
According to the RPM doc, the `BuildRequires:` part should not be printed when using the script for `%generate_buildrequires`

> %generate_buildrequires (since rpm >= 4.15)
>
>This optional script can be used to determine BuildRequires dynamically. If present it is executed after %prep and can though access the unpacked and patched sources. The script must print the found build dependencies to stdout in the same syntax as used after BuildRequires: one dependency per line.
>
>rpmbuild will then check if the dependencies are met before continuing the build. If some dependencies are missing a package with the .buildreqs.nosrc.rpm postfix is created, that - as the name suggests - contains the found build requires but no sources. It can be used to install the build requires and restart the build.
>
>On success the found build dependencies are also added to the source package. As always they depend on the exact circumstance of the build and may be different when bulding based on other packages or even another architecture.